### PR TITLE
 Added jaxb dependencies for Java 11 compatibility - CLM-12408 CLM-12430

### PR DESCRIPTION
**Pull request by mealingr**
Story: https://issues.sonatype.org/browse/CLM-12408
Task: https://issues.sonatype.org/browse/CLM-12430
Build: https://jenkins.zion.aws.s/job/integrations/job/jenkins/job/sonatype-nexus-platform-plugin-feature/job/CLM-12430_jaxb/lastBuild

To be built and merged after https://github.com/jenkinsci/nexus-platform-plugin/pull/71

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
   <properties>
     <enforcer.skip>true</enforcer.skip> <!-- TODO numerous requireUpperBoundDeps, some probably indicative of real problems -->
     <jenkins.version>2.7</jenkins.version>
-    <findbugs-maven-plugin.version>3.0.2</findbugs-maven-plugin.version>
+    <findbugs-maven-plugin.version>3.0.5</findbugs-maven-plugin.version>
```
Comment (mealingr) [inline pom.xml]: Temporary until https://github.com/jenkinsci/nexus-platform-plugin/pull/71 is merged

File: pom.xml [hunk 2]
```diff
@@ -297,6 +314,11 @@
     </resources>
     <pluginManagement>
       <plugins>
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>animal-sniffer-maven-plugin</artifactId>
+          <version>1.17</version>
+        </plugin>
```
Comment (mealingr) [inline pom.xml]: Temporary until https://github.com/jenkinsci/nexus-platform-plugin/pull/71 is merged

File: pom.xml [hunk 3]
```diff
@@ -286,6 +286,23 @@
       <artifactId>wiremock-standalone</artifactId>
       <scope>test</scope>
     </dependency>
+
+    <!-- For Java 11-->
+    <dependency>
+      <groupId>javax.xml.bind</groupId>
+      <artifactId>jaxb-api</artifactId>
```
Comment (bentmann) [inline pom.xml]: What here uses JAXB?

Comment (mealingr) [reply pom.xml]: > What here uses JAXB?

Hmm I thought I was getting errors without jaxb for Java 11, testing again as I may have been mistaken...

Comment (mealingr) [reply pom.xml]: > Hmm I thought I was getting errors without jaxb for Java 11, testing again as I may have been mistaken...

I think I was mistaken, it seems to build fine without these jaxb dependencies https://jenkins.zion.aws.s/job/integrations/job/jenkins/job/sonatype-nexus-platform-plugin-feature/job/CLM-12051_openjdk11_min/ closing this PR

**Comment by chriscarpenter12** [comment]
Running under Java 11 we're seeing and error requiring a JAXB dependency. Why was this issue closed? Using the Jenkins [JAXB plugin](https://plugins.jenkins.io/jaxb/) doesn't seem to make a difference. Reverting Jenkins to use Java 8 works, but we'd like to continue using Java 11.

nexus-error

**Comment by mealingr** [comment]
> Why was this issue closed?

@chriscarpenter12 The goal of this was to build the `nexus-platform-plugin` with Java 11 (OpenJDK 11), which at the time was successful without needing the changes from this PR as per this comment https://github.com/jenkinsci/nexus-platform-plugin/pull/75#discussion_r275783258

> Running under Java 11 we're seeing and error requiring a JAXB dependency

This closed PR probably isn't the best place to discuss your issue, it may be related but it's also over a year old so I'd open a new JIRA issue for this under the Integrations project. If you have the information it'd probably also be helpful to have the full stack trace as well as the exact NXRM3, Jenkins, and plugin versions you used at the time

**Comment by chriscarpenter12** [comment]
I came across this issue while searching for the issue I had. Can you point me toward the Integrations project so I can open an issue with all the relevant details?

**Comment by mealingr** [comment]
> I came across this PR while searching for the issue I had. Can you point me toward the Integrations project so I can open an issue with all the relevant details?

Sure if you go to JIRA https://issues.sonatype.org/ and "Create" in the top bar you should hopefully be able to set the "Project" to "Integrations (INT)", probably want to set it as a bug, something like https://issues.sonatype.org/browse/INT-3837

**Comment by chriscarpenter12** [comment]
Can't set Project to Integrations.

image

Also don't have permission to view your example issue.

image

**Comment by mealingr** [comment]
> Can't set Project to Integrations.

Ah ok, in that case you could either contact an EM to give you access, or I'd be happy to create it for you if you slack me the details, but the former is probably best so you can monitor the issue and see updates, perhaps @nblair or @timlevett can help

**Comment by mpuglin** [comment]
@chriscarpenter12  As @mealingr stated, please create a support ticket vis the support.sonatype.com website.  Supplying your log will be very helpful to reviewing this more closely.  If you are unable to open a ticket for any reason, please comment here.   I will work with you to get one created.

**Comment by chriscarpenter12** [comment]
@mpuglin & @mealingr Thanks to both of you for the help! I was able to get a ticket filed out: `Request #48444`

**Comment by chriscarpenter12** [comment]
@mpuglin Support pointed me back here to the issues on Github as I don't have a valid support contract... I opened an [issue here](https://github.com/jenkinsci/nexus-platform-plugin/issues/133).

**Pull request closed without merging**